src/components/Tag/Tag.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { MdClose } from 'react-icons/md';
+import { bem } from '../../utils';
+import { Text } from '../Text';
+
+import styles from './Tag.scss';
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+    /** Value that helps us to understand if the component is currently selected or not. */
+    isSelected?: boolean;
+    /** Color is assigned to a background color of a wrapper */
+    bgColor?: string;
+    /** Max-width of a component */
+    maxWidth?: string;
+    /** Size of the text */
+    size?: 'small' | 'normal' | 'large';
+    /** Callback, that is fired when a user clicks on a delete icon */
+    onDelete?: () => void;
+    /** Callback, that is fired when a user clicks on an element */
+    onClick?: () => void;
+}
+
+const { block, elem } = bem('Tag', styles);
+
+export const Tag: React.FC<Props> = (props) => {
+    const { children, bgColor, maxWidth, size, onDelete, onClick, isSelected, ...rest } = props;
+
+    // Generate title for children that are plain text (without tags)
+    // If there is something different from the string (JSX) - children will be of object type.
+    const areChildrenString = typeof children === 'string';
+
+    const handleDeleteClick = (e: React.MouseEvent<HTMLElement>) => {
+        e.stopPropagation();
+
+        onDelete?.();
+    };
+
+    return (
+        <div
+            {...block({ selected: isSelected, clickable: !!onClick, ...rest })}
+            {...(onClick && { onClick, tabIndex: 0, role: 'button' })}
```

Comment:
Sorry, one more thing. The clickable Tag is not keyboard accessible. I guess the linter doesn't complain because the conditional render of the onClick attribute. 
We could achieve keyboard accessibility in 2 ways. 
1. add a onKeyPress handler, which calls onClick if the key pressed was enter. Can search a codebase  for `ENTER_KEY` for examples if you need.
2. when onClick is defined, render the top as a button not as div. But then we have button within button, not exactly semantic HTML. Anyway you would do it with  something like this:
```
const As = onClcik ? 'button' : 'div'; // Note the capital A which allows JSX to recognise it as valid
...
return <As {...rest} {...block(...)}>...</As>
```

So I guess the first solution is better, but leave it to you.

Reply:
Implemented.